File: Human_written_code/Code_with_comments/CPP/Q12.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int n = 3; // Position in the Fibonacci sequence

    int prev2 = 1; // (n-2)th Fibonacci number
    int prev = 1;  // (n-1)th Fibonacci number

    // Compute the nth Fibonacci number
    for (int i = 2; i <= n; i++) {
        int cur_i = prev2 + prev;
        prev2 = prev;
        prev = cur_i;
    }
    
    cout << prev; // Output the nth Fibonacci number
    return 0;
}

File: Human_written_code/Code_with_comments/Python/Q12.py
def main():
    n = 3  # The position in the Fibonacci sequence to calculate
    prev2 = 1  # Fibonacci number at position n-2
    prev = 1    # Fibonacci number at position n-1

    # Compute the Fibonacci number at position n
    for i in range(2, n+1):
        cur_i = prev2 + prev  # Current Fibonacci number
        prev2 = prev          # Update prev2 to the previous value
        prev = cur_i          # Update prev to the current value

    print(prev)  # Output the Fibonacci number at position n

if __name__ == "__main__":
    main()
